### DISCUSSION TITLE
Some scan pdfs' text is wrong, i think the text is gotten from ocr, so how can l get the original background image?

### DISCUSSION THREAD

**Discussion by njustczr** [pymupdf/PyMuPDF#2132] [Q&A]

**Comment by JorjMcKie** [comment]
This has no description at all.
The title suggests you may have wanted to ask a question in "Discussions".
If so, please open a post there - and with sufficient background please.

**Comment by njustczr** [comment]
> This has no description at all. The title suggests you may have wanted to ask a question in "Discussions". If so, please open a post there - and with sufficient background please.

the pdf is 
![image](https://user-images.githubusercontent.com/30426320/208062995-8b63ef53-8dca-4d93-a4b2-0a448481ced7.png)
but i copy the text is "第丸章 肘足虫"  
the text is gotten by OCR, so if i can get the scan background image

**Reply by JorjMcKie** [reply]
You can either use a package like OCRmyPDF to force-rescan the document, or use PyMuPDF's built-in OCR feature.
In both cases you should make sure to specify the presence of language Chinese.

You can also look at the page's images `page.get_images()` and extract them via `doc.extract_image()`. But I recommend trying to re-OCR the page instead. Use a large dpi to improve the recognition quality.

**Reply by njustczr** [reply]
I have tried the following code to save the background image,but i can only get a white image.
    # doc = fitz.open(pdf_path)
    # page = doc[10]
    # xrefs = page.get_images()
    # print(xrefs)
    # idx = 0
    # for xref in xrefs:
    #     idx += 1
    #     pix = fitz.Pixmap(doc, xref[0])
    #     if pix.n < 5:
    #         pix.save(str(idx) + '.png')
    #     else:
    #         pix1 = fitz.Pixmap(fitz.csRGB, pix)
    #         pix1.save(str(idx) + ".png")
    #         pix1 = None
    #     pix = None

**Reply by njustczr** [reply]
> You can either use a package like OCRmyPDF to force-rescan the document, or use PyMuPDF's built-in OCR feature. In both cases you should make sure to specify the presence of language Chinese.
> 
> You can also look at the page's images `page.get_images()` and extract them via `doc.extract_image()`. But I recommend trying to re-OCR the page instead. Use a large dpi to improve the recognition quality.

thanks
